skpr/cognito-auth
# Refactor to add keychain cache for tokens

#### What does this PR do?

Adds support for storing oauth2 token and aws credentials in a native keychain.

#### How should this be manually tested?

Add a `--creds-store` parameter to the login command:
```
./bin/cognito_auth_darwin_amd64 userpool login --username=test4 --creds-store=native
```

Open Keychain Access  (MacOS) and check 2 new entries were added: oauth tokens and aws creds.

#### Any background context you want to provide?
 
https://medium.com/@calavera/stop-saving-credential-tokens-in-text-files-65e840a237bb

#### What are the relevant tickets?
 
#### Screenshots (if appropriate)
 
#### Questions:
 
##### Does any external documentation require updating?

##### Does this changeset require specific versions of supporting software?
(i.e. Go / Terraform / Docker)
